src/generators/utils/parse.ts

```diff
@@ -47,3 +55,132 @@ export function parseSitesFixesConfig<T extends SiteProps>(text: string, options
+
+    function processBlock(recordStart: number, recordEnd: number, index: number) {
+        // TODO: more formal definition of URLs and delimiters
+        const block = text.substring(recordStart, recordEnd);
+        const lines = block.split('\n');
+        const commandIndices: number[] = [];
+        lines.forEach((ln, i) => {
+            if (ln.match(/^[A-Z]+(\s[A-Z]+){0,2}$/)) {
+                commandIndices.push(i);
+            }
+        });
+
+        if (commandIndices.length === 0) {
+            return;
+        }
+
+        const urls = parseArray(lines.slice(0, commandIndices[0]).join('\n'));
+        for (const url of urls) {
+            const domain = getDomain(url);
+            if (isFullyQualifiedDomain(domain)) {
+                if (!domains[domain]) {
+                    domains[domain] = index;
+                } else if (typeof domains[domain] === 'number' && domains[domain] !== index) {
+                    domains[domain] = [(domains[domain] as number), index];
+                } else if (typeof domains[domain] === 'object' && !((domains[domain] as number[]).includes(index))) {
+                    (domains[domain] as number[]).push(index);
+                }
+                continue;
+            }
+
+            if (!domains[domain]) {
+                domainPatterns[domain] = index;
+            } else if (typeof domains[domain] === 'number' && domains[domain] !== index) {
+                domainPatterns[domain] = [(domainPatterns[domain] as number), index];
+            } else if (typeof domains[domain] === 'object' && !((domains[domain] as number[]).includes(index))) {
+                (domainPatterns[domain] as number[]).push(index);
+            }
+        }
+        offsets.push([recordStart, recordEnd - recordStart]);
+    }
+
+    let recordStart = 0;
+    // Delimiter between two blocks
+    const delimiterRegex = /\r?\s*={2,}\s*\r?/gm;
+    let delimiter: RegExpMatchArray;
+    let count = 0;
+    while ((delimiter = delimiterRegex.exec(text))) {
+        const nextDelimiterStart = delimiter.index;
+        const nextDelimiterEnd = delimiter.index + delimiter[0].length;
+
+        processBlock(recordStart, nextDelimiterStart, count);
+
+        recordStart = nextDelimiterEnd;
+        count++;
+    }
+    processBlock(recordStart, text.length, count);
+
+    return {offsets: encodeOffsets(offsets), domains, domainPatterns, cache: {}};
+}
+
+export function parseSiteFixConfig<T extends SiteProps>(text: string, options: SitesFixesParserOptions<T>, recordStart: number, recordEnd: number): T {
+    const block = text.substring(recordStart, recordEnd);
+    return parseSitesFixesConfig<T>(block, options)[0];
+}
+
+export function getSitesFixesFor<T extends SiteProps>(url: string, text: string, index: SitePropsIndex<T>, options: SitesFixesParserOptions<T>): T[] {
+    const records: T[] = [];
+    let recordIds: number[] = [];
+    const domain = getDomain(url);
```

Comment:
Ah stupid stupid me, `url` is the full blown URL and not a sanitized one. I think you can copy paste that sanatizeInput function over and use it here and probably as well for the other one.

Reply:
Can we just use the native `URL` constructor? E.g.,`new URL(url).hostname`.

Reply:
I changed `getDomain()` a bit to accomodate incomplete URLs:
```js
function getDomain(url: string) {
    try {
        return (new URL(url)).hostname.toLowerCase();
    } catch (error) {
        return url.split('/')[0].toLowerCase();
    }
}
```